ycmd/tests/clang/flags_test.py

```diff
@@ -604,6 +604,34 @@ def CompilationDatabase_HeaderFileHeuristicNotFound_test():
         [] )
 
 
+def CompilationDatabase_ExplicitHeaderFileEntry_test():
+  with TemporaryClangTestDir() as tmp_dir:
+    # Have an explicit header file entry which should take priority over the
+    # corresponding source file
+    compile_commands = [
+      {
+        'directory': tmp_dir,
+        'command': 'clang++ -x c++ -I. -I/absolute/path -Wall',
+        'file': os.path.join( tmp_dir, 'test.cc' ),
+      },
+      {
+        'directory': tmp_dir,
+        'command': 'clang++ -I/absolute/path -Wall',
+        'file': os.path.join( tmp_dir, 'test.h' ),
+      },
+    ]
+    with TemporaryClangProject( tmp_dir, compile_commands ):
+      assert_that(
+        flags.Flags().FlagsForFile(
+          os.path.join( tmp_dir, 'test.h' ),
+          add_extra_clang_flags = False ),
+        contains( 'clang++',
+                  '-x',
+                  'c++',
```

Comment:
Am I losing it or should his flag NOT be present? Or are we artificially adding it due to 'clang++'

Reply:
https://github.com/Valloric/ycmd/blob/2d79555c70e3bc6e5bda94f3029f1366fa609d32/ycmd/completers/cpp/flags.py#L359